### ISSUE TITLE
Alfresco db issue

### ISSUE THREAD

**Issue by SanjayBandhniya** [Alfresco/alfresco-docker-installer#202]
I am using docker alfresco installer for ubuntu.
I have created volume using *.sh script still I am getting below error.

alfresco-1            | Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'tenantService' defined in URL [jar:file:/usr/local/tomcat/webapps/alfresco/WEB-INF/lib/alfresco-repository-21.14.jar!/alfresco/mt/mt-context.xml]: Cannot resolve reference to bean 'tenantAdminDAO' while setting bean property 'tenantAdminDAO'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'tenantAdminDAO' defined in class path resource [alfresco/dao/dao-context.xml]: Cannot resolve reference to bean 'repoSqlSessionTemplate' while setting bean property 'sqlSessionTemplate'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'repoSqlSessionTemplate' defined in class path resource [alfresco/ibatis/ibatis-context.xml]: Cannot resolve reference to bean 'repoSqlSessionFactory' while setting constructor argument; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'repoSqlSessionFactory' defined in class path resource [alfresco/ibatis/ibatis-context.xml]: Cannot resolve reference to bean 'dialectResourceLoader' while setting bean property 'resourceLoader'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'dialectResourceLoader' defined in class path resource [alfresco/ibatis/ibatis-context.xml]: Cannot create inner bean 'org.springframework.beans.factory.config.PropertyPathFactoryBean#3da7610a' of type [org.springframework.beans.factory.config.PropertyPathFactoryBean] while setting bean property 'dialectClass'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.springframework.beans.factory.config.PropertyPathFactoryBean#3da7610a' defined in class path resource [alfresco/ibatis/ibatis-context.xml]: Initialization of bean failed; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'dialect': FactoryBean threw exception on object creation; nested exception is org.springframework.jdbc.CannotGetJdbcConnectionException: Failed to obtain JDBC Connection; nested exception is java.sql.SQLException: Cannot create PoolableConnectionFactory (The connection attempt failed.)
alfresco-1            |         at org.springframework.beans.factory.support.BeanDefinitionValueResolver.resolveReference(BeanDefinitionValueResolver.java:342) ~[spring-beans-5.3.27.jar:5.3.27]

Any idea what could be the issue?

**Comment by aborroy** [comment]
Verify if the database (postgres service) is up & ready. Spoiler: it's not.

**Comment by SanjayBandhniya** [comment]
DB is already up.
Database also getting created.

**Comment by aborroy** [comment]
Try with the following sequence:

```
docker compose up postgres
docker compose up
```

So you provide some time to postgres to start properly.

**Comment by SanjayBandhniya** [comment]
> Try with the following sequence:
> 
> ```
> docker compose up postgres
> docker compose up
> ```
> 
> So you provide some time to postgres to start properly.

Tried same but no luck.

**Comment by aborroy** [comment]
You may try starting only `postgres` service and verify it works as expected, using some tool like `psql` to verify database and user `alfresco` is running.

**Comment by SanjayBandhniya** [comment]
> You may try starting only `postgres` service and verify it works as expected, using some tool like `psql` to verify database and user `alfresco` is running.

Yes. Its working. I am able to inspect container and also able to check alfresco db also created.

**Comment by aborroy** [comment]
May you add your `docker-compose.yml` content to this issue?

**Comment by SanjayBandhniya** [comment]
> May you add your `docker-compose.yml` content to this issue?

I am using same docker file in another machine where it is working fine.
From working machine I copied entire setup with db and data to new machine but still facing same error.

[docker-compose.txt](https://github.com/user-attachments/files/16426145/docker-compose.txt)

**Comment by aborroy** [comment]
The error seems to be that repository is not able to connect to postgres. Are there additional errors in the log?

**Comment by SanjayBandhniya** [comment]
> The error seems to be that repository is not able to connect to postgres. Are there additional errors in the log?

[error.txt](https://github.com/user-attachments/files/16426268/error.txt)
This is entire error log.

**Comment by aborroy** [comment]
`Caused by: java.net.SocketTimeoutException: Connect timed out`

Repository is not connecting to `postgres:5432`. You may try checking connectivity from inside repository container to identify to cause.
